integration/query_array_test.go

```diff
@@ -50,31 +50,12 @@ func TestQueryArraySize(t *testing.T) {
 		expectedIDs []any
 		err         *mongo.CommandError
 	}{
-		"int32": {
-			filter:      bson.D{{"v", bson.D{{"$size", int32(2)}}}},
-			expectedIDs: []any{"array-two"},
-		},
-		"int64": {
-			filter:      bson.D{{"v", bson.D{{"$size", int64(2)}}}},
-			expectedIDs: []any{"array-two"},
-		},
-		"float64": {
-			filter:      bson.D{{"v", bson.D{{"$size", float64(2)}}}},
-			expectedIDs: []any{"array-two"},
-		},
-		"Zero": {
-			filter:      bson.D{{"v", bson.D{{"$size", 0}}}},
-			expectedIDs: []any{"array-empty"},
-		},
 		"NegativeZero": {
 			filter:      bson.D{{"v", bson.D{{"$size", math.Copysign(0, -1)}}}},
 			expectedIDs: []any{"array-empty"},
 		},
-		"NotFound": {
-			filter:      bson.D{{"v", bson.D{{"$size", 4}}}},
-			expectedIDs: []any{},
-		},
 		"InvalidType": {
+			// TODO: move to compat https://github.com/FerretDB/FerretDB/issues/1539
```

Comment:
Please mark this PR as ready for review soon, and then investigate that issue, as we clearly have code for that: https://github.com/FerretDB/FerretDB/blob/8029b86ae0a2a2b2f4a22adf8e9e79d960c131a2/internal/handlers/common/filter.go#L712-L721

Reply:
Yeah let me do that soon. I will only address array query compat test in this PR.

Reply:
I updated the issue with example https://github.com/FerretDB/FerretDB/issues/1539
This only fails on scalar values.